parsers/AR.py

```diff
@@ -23,957 +22,111 @@
-      '3%ADa/Operativos/VisorReporteSinComDesp_minimal.aspx?hora=0&titulo=Des' \
-      'pacho%20Generacion%20Termica&reportPath=http://lauzet:5000/MemNet1/Report' \
-      'ingServices/DespachoGeneracionTermicaLP.rdl--0--Despacho+Generaci%c3%b3n+T%c3%a9rmica'
-
-turl = ('http://portalweb.cammesa.com/Reserved.ReportViewerWebControl.'
-        'axd?Culture=3082&UICulture=3082&ReportStack=1'
-        '&OpType=ReportArea&Controller=ClientController'
-        'ctl00_ctl04_g_a581304b_aafc_4818_a4a1_e96f27a22246_ctl00_RepViewer'
-        '&ZoomMode=Percent&ZoomPct=100&ReloadDocMap='
-        'true&SearchStartPage=0&LinkTarget=_top')
-
-hurl = ('http://portalweb.cammesa.com/memnet1/Pages/Informes%20por%20Categor'
-        '%C3%ADa/Operativos/VisorReportesSinCom_minimal.aspx?hora=0&'
-        'titulo=Despacho%20Generacion%20Hidraulica&reportPath='
-        'http://lauzet:5000/MemNet1/ReportingServices/'
-        'DespachoGeneracionHidraulica.rdl--0--Despacho+Generaci%c3%b3n+Zona+'
-        'Hidr%c3%a1ulica')
-
-thurl = ('http://portalweb.cammesa.com/Reserved.ReportViewerWebControl.'
-         'axd?Culture=3082&UICulture=3082&ReportStack=1'
-         '&OpType=ReportArea&Controller=ClientController'
-         'ctl00_ctl04_g_966166c3_db78_453e_9a34_83d2bb263ee4_''ctl00_RepViewer'
-         '&ZoomMode=Percent&ZoomPct=100&ReloadDocMap='
-         'true&SearchStartPage=0&LinkTarget=_top')
-
-cammesa_url = 'http://portalweb.cammesa.com/default.aspx'
-
-tie_mapping = {'CL-SEN': "position:absolute; top:349; left:585",
-               'PY': "position:absolute; top:67; left:649",
-               'UY_1': "position:absolute; top:203; left:533",
-               'UY_2': "position:absolute; top:226; left:515"
-               }
-
-
-def webparser(req):
-    """Takes content from webpage and returns all text as a list of strings"""
-
-    soup = BeautifulSoup(req.content, 'html.parser')
-    figs = soup.find_all("div", class_="r11")
-    data_table = [unicode(tag.get_text()) for tag in figs]
-
-    return data_table
-
-
-def fetch_price(zone_key='AR', session=None, target_datetime=None, logger=logging.getLogger(__name__)) -> dict:
-    """Requests the last known power price of a given country."""
     if target_datetime:
         raise NotImplementedError('This parser is not yet able to parse past dates')
-    s = session or requests.Session()
-    price_req = s.get(cammesa_url)
-    psoup = BeautifulSoup(price_req.content, 'html.parser')
-    find_price = psoup.find('td', class_="cssFuncionesLeft", align="left")
 
-    try:
-        price_text = find_price.getText()
+    current_session = session or requests.session()
+
+    non_renewables_production: Dict[str, dict] = non_renewables_production_mix(zone_key, current_session)
+    renewables_production: Dict[str, dict] = renewables_production_mix(zone_key, current_session)
+    
+    full_production_list = [
+        {
+            'datetime': arrow.get(datetime_tz_ar).to('UTC').datetime,
```

Comment:
Is `.to('UTC')` necessary here? I'm not sure if the `datetime` expects the time to be in a specific timezone.

Reply:
It should be in UTC, so this is good :)